pyroll/zouhar_contact.py

```diff
@@ -70,11 +75,16 @@ def contact_area(self: RollPass.Roll):
             and
             rp.zouhar_contact_in_width is not None
     ):
+        if "3fold" in rp.classifiers:
```

Comment:
Maybe you can avoid such checks if you would define a new hook function for TRP only. Refactor the main logic to a simple function which is called from the respective hook impl. Then you would not need the core changes (removal of TRP default hook impls there).